Go/Go.sublime-syntax

```diff
@@ -401,7 +401,7 @@ contexts:
 
   match-imaginary:
     # Mandatory integer, optional fraction, optional exponent, mandatory imaginary
-    - match: \d+(?:(\.)\d+)?(?:({{exponent}})\d+)?(i)
+    - match: \d+(?:_\d+)*(?:(\.)\d+(?:_\d+)*)?(?:({{exponent}})\d+(?:_\d+)*)?(i)
```

Comment:
Would you considder a variable for `(?:\d+(?:_\d+)*)` ?

Recently found Python doing so and found it helpful with regards to readability. I therefore used and extended its use in https://github.com/sublimehq/Packages/pull/2136

```
  ddigit: '\d+(?:_\d+)*'
  hdigit: '\h+(?:_\h+)*'
```

The match would look like

```
 - match: '{{ddigit}}(?:(\.){{ddigit}})?(?:({{exponent}}){{ddigit}})?(i)'
```

Reply:
A further step might be to update the `exponent` variable, which would then effect many more places, of cource.

Example:

```
  exponent: '(?:([Ee])[-+]?{{ddigit}})'
```

The match above would then look like

```
 - match: '{{ddigit}}(?:(\.){{ddigit}})?{{exponent}}?(i)'
```

This is how python handles it.

_Note:_

This change excludes the sign of the exponent from the punctuation scope, which I find is wrong. If so only `e` should be scoped as separator, but I wonder whether it should be scoped as punctuation at all.

Reply:
Note that the variables should look more like this (plural s and non-capturing group):
```
ddigits: (?:\d+(?:_\d+)*)
hdigits: (?:\h+(?:_\h+)*)
```

Reply:
I added the `ddigits` and `hdigits` variables. I agree they improve readability.